src/main/java/org/tron/core/net/node/NodeImpl.java

```diff
@@ -748,7 +749,14 @@ private void onHandleSyncBlockChainMessage(PeerConnection peer, SyncBlockChainMe
     }
 
     if (blockIds.isEmpty()) {
-      peer.setNeedSyncFromUs(false);
+      if (CollectionUtils.isNotEmpty(summaryChainIds) && summaryChainIds.get(0).getNum() < del
+          .getHeadBlockId().getNum()) {
```

Comment:
when peer's number id lower than our solid block number , It's really sync fail, in this case , it maybe can sync again

Reply:
modify use LatestBlockHeaderNumber compare